arangod/Network/Utils.cpp

```diff
@@ -313,15 +313,12 @@ std::string fuerteStatusToArangoErrorMessage(fuerte::Response const& res) {
   return fuerte::status_code_to_string(res.statusCode());
 }
 
-void addSourceHeader(fuerte::Request& req) {
+void addSourceHeader(consensus::Agent* agent, fuerte::Request& req) {
   auto state = ServerState::instance();
   if (state->isCoordinator() || state->isDBServer()) {
     req.header.addMeta(StaticStrings::ClusterCommSource, state->getId());
-  } else if (state->isAgent()) {
-    auto agent = AgencyFeature::AGENT;
-    if (agent != nullptr) {
-      req.header.addMeta(StaticStrings::ClusterCommSource, "AGENT-" + agent->id());
-    }
+  } else if (state->isAgent() && agent != nullptr) {
+    req.header.addMeta(StaticStrings::ClusterCommSource, "AGENT-" + agent->id());
   }
```

Comment:
While this mimics the previous behavior, I am wondering if this distinction between coordinators/dbservers and agents is actually necessary, and if we shouldn't always add the value returned by `state->getId()`. Need to check what this returns for agents!